src/cargo/ops/cargo_add/mod.rs

```diff
@@ -596,12 +600,68 @@ fn populate_dependency(mut dependency: Dependency, arg: &DepOp) -> Dependency {
     dependency
 }
 
+pub struct DependencyEx {
```

Comment:
I'm not sold on `DependencyEx`. The name doesn't make sense to me. This appears to be only used for displaying with features, so maybe `DisplayDependency` or something similar.

Reply:
`Ex` / `Ext` are common suffixes for "extension", conveying that this is an extension on the `Dependency` type.  This dates past long before `ResultExt` type names in Rust.  iirc a lot of old Microsoft APIs use it.

That said, I named it that not because I liked it but because I needed a name and wanted to move forward.

`DisplayDependency` implies the purpose is for the `Display`.

I've gone ahead and renamed it `DependencyUI` even though that still isn't quite accurate, bit is at least a little more specific.